worlds/pokemon_crystal/__init__.py

```diff
@@ -0,0 +1,333 @@
+
+        total_badges = max(self.options.elite_four_badges.value, self.options.red_badges.value)
+        add_badges = []
+        # Extra badges to add to the pool in johto only
+        if self.options.johto_only and total_badges > 8:
+            kanto_badges = [item_id for item_id, item_data in crystal_data.items.items() if
+                            "KantoBadge" in item_data.tags]
+            self.random.shuffle(kanto_badges)
+            add_badges = kanto_badges[:total_badges - 8]
+
+        traps_pool = []
+        traps_pool += ["Phone Trap"] * self.options.phone_trap_weight.value
+        traps_pool += ["Sleep Trap"] * self.options.sleep_trap_weight.value
+        traps_pool += ["Poison Trap"] * self.options.poison_trap_weight.value
+        traps_pool += ["Burn Trap"] * self.options.burn_trap_weight.value
+        traps_pool += ["Freeze Trap"] * self.options.freeze_trap_weight.value
+        traps_pool += ["Paralysis Trap"] * self.options.paralysis_trap_weight.value
+
+        total_trap_weight = len(traps_pool)
+
+        def get_random_trap():
+            return self.create_item(self.random.choice(traps_pool))
+
+        default_itempool = []
+
+        for location in item_locations:
+            item_code = location.default_item_code
+            if item_code > 0 and get_item_classification(item_code) != ItemClassification.filler:
+                if item_code in crystal_data.tm_replace_map and self.options.randomize_tm_moves:
+                    default_itempool += [self.create_item_by_code(item_code + 256)]
+                else:
+                    default_itempool += [self.create_item_by_code(item_code)]
+            elif len(add_badges):
+                default_itempool += [self.create_item_by_code(add_badges.pop())]
+            elif self.random.randint(0, 100) < total_trap_weight:
+                default_itempool += [get_random_trap()]
+            elif item_code == 0:  # item is NO_ITEM, trainersanity checks
+                default_itempool += [self.create_item_by_const_name(get_random_filler_item(self.random))]
+            else:
+                default_itempool += [self.create_item_by_code(item_code)]
+
+        self.multiworld.itempool += default_itempool
+
+    def set_rules(self) -> None:
+        set_rules(self)
+
+    def pre_fill(self) -> None:
+        if self.options.randomize_badges.value == RandomizeBadges.option_shuffle:
+            badge_locs = [loc for loc in self.multiworld.get_locations(self.player) if "Badge" in loc.tags]
+            badge_items = [self.create_item_by_code(loc.default_item_code) for loc in badge_locs]
+            if self.options.early_fly:
+                # take one of the 3 early badge locations, set it to storm badge
+                storm_loc = self.random.choice([loc for loc in badge_locs if "EarlyBadge" in loc.tags])
+                storm_badge = [item for item in badge_items if item.name == "Storm Badge"][0]
```

Comment:
```suggestion
                storm_badge = next(item for item in badge_items if item.name == "Storm Badge")
```
Bit of a nitpick, since `badge_items` is <20 items, but this version short circuits.